### ISSUE TITLE
PATH in child process includes Windows Terminal Preview installation directory, if started via Win+R wt.exe

### ISSUE THREAD

**Issue by KalleOlaviNiemitalo** [microsoft/terminal#7204]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.388]
Windows Terminal version (if applicable): Windows Terminal Preview 1.2.2022.0
```

# Steps to reproduce

1. Enable the `wt.exe` app execution alias of Windows Terminal Preview.
2. Press <kbd>Windows</kbd>+<kbd>R</kbd> to open the <samp>Run</samp> dialog box.
3. Type `wt.exe` and press <kbd>Enter</kbd> to start Windows Terminal Preview.
4. In Windows Terminal Preview, start the Command Prompt profile.
5. In the Command Prompt session, type `PATH` and press <kbd>Enter</kbd>.

# Expected behavior

The displayed value of PATH does not include the directory where Windows Terminal Preview was installed.

# Actual behavior

`C:\Program Files\WindowsApps\Microsoft.WindowsTerminalPreview_1.2.2022.0_x64__8wekyb3d8bbwe` is the first directory in PATH.

# Notes

This is presumably caused by the Registry values that were created by AppXSvc when Windows Terminal Preview was installed:

```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\wt.exe]
@="C:\\Program Files\\WindowsApps\\Microsoft.WindowsTerminalPreview_1.2.2022.0_x64__8wekyb3d8bbwe\\wt.exe"
"Path"="C:\\Program Files\\WindowsApps\\Microsoft.WindowsTerminalPreview_1.2.2022.0_x64__8wekyb3d8bbwe"

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\wt.exe]
@="C:\\Program Files\\WindowsApps\\Microsoft.WindowsTerminalPreview_1.2.2022.0_x64__8wekyb3d8bbwe\\wt.exe"
"Path"="C:\\Program Files\\WindowsApps\\Microsoft.WindowsTerminalPreview_1.2.2022.0_x64__8wekyb3d8bbwe"
```

According to [Application Registration](<https://docs.microsoft.com/windows/win32/shell/app-registration#using-the-app-paths-subkey> "Application Registration - Win32 apps | Microsoft Docs"), ShellExecuteEx prepends the data of the "Path" Registry value to the PATH environment variable of the new process. That seems unnecessary here, as selecting Windows Terminal Preview from the Start menu does not add to PATH.

Because non-packaged processes of users have no access to the "C:\Program Files\WindowsApps" directory, this addition to PATH perhaps does not cause problems. However, if `wt.exe` is run as administrator, then the PATH addition could cause a process in the terminal session to load DLLs from the installation directory of Windows Terminal Preview. (If a packaged application is run within the terminal session, then that too would have access to the directory, but [Search Order for Windows Store apps](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order#search-order-for-windows-store-apps) looks like the PATH environment variable will not affect DLL loading in that case.)

Related to <https://github.com/microsoft/terminal/pull/6860>, <https://github.com/microsoft/terminal/issues/6748>, <https://github.com/microsoft/terminal/issues/7188>. Found while searching for possible causes of <https://github.com/microsoft/terminal/issues/7195>.

**Comment by DHowett** [comment]
So, a few notes.

* I think this is harmless. There's no executable in our package that the user shouldn't be able to launch, and nothing that will incorrectly eclipse things that _are_ on their `PATH`s.
* We don't have any control over this.
* I can't find this entry in `PATH` when I elevate wt.exe
* Packaged applications don't follow the traditional DLL search order, so I don't think it likely that they'll inherit WT's libraries

**Comment by DHowett** [comment]
Incidentally, this and the last seven issues are the first (and second through seventh) times I've ever heard of `App Paths`, and I didn't know the system was maintaining it on our behalf.

**Comment by KalleOlaviNiemitalo** [comment]
>     * I can't find this entry in `PATH` when I elevate wt.exe

How do you elevate it? If I type `wt.exe` in the <samp>Run</samp> dialog box and press Ctrl+Shift+Enter, then the WT installation directory does appear in PATH in the Command Prompt session.

**Comment by DHowett** [comment]
Yep, using <kbd>Ctrl+Shift+Enter</kbd> from the Run dialog.

**Comment by KalleOlaviNiemitalo** [comment]
> We don't have any control over this.

I imagine Windows Terminal could explicitly delete its installation directory from PATH in the environment block that it passes to CreateProcess.

Or use CreateEnvironmentBlock as suggested in <https://github.com/microsoft/terminal/issues/1125>, instead of copying PATH and other variables from the environment of the WindowsTerminal.exe process.

> Yep, using Ctrl+Shift+Enter from the Run dialog.

That's strange… Are you using the same version of Windows 10? Does the PATH change even get to the WindowsTerminal.exe process? Process Monitor filtering with Operation = Process Start is handy for checking this.

**Comment by KalleOlaviNiemitalo** [comment]
<https://github.com/microsoft/terminal/pull/7243> did not fix this for the following child processes:

* the editor that Windows Terminal starts for editing `settings.json` (and presumably `defaults.json`)
* `C:\WINDOWS\system32\wsl.exe --list`
* Web browser processes started by the Feedback menu item or by hyperlinks on the About screen

According to <https://github.com/microsoft/terminal/issues/7460>, the extra entry in `Path` interferes with the loading of `vcruntime140.dll` from other directories in `Path`, if `C:\Windows\System32\VCRUNTIME140.dll` does not exist.

Please reopen.

**Comment by DHowett** [comment]
_Huh._

**Comment by JustinGrote** [comment]
@DHowett  not sure if this is related but when I do `Win+R` then `wt.exe`, currently getting this:
![image](https://user-images.githubusercontent.com/15258962/94107592-adeb1880-fdf2-11ea-8930-e2afed95d8d0.png)

I've tried lots of combinations of my PATH variable to no avail.

**Comment by KalleOlaviNiemitalo** [comment]
Retested with Windows Terminal Preview 1.12.3472.0 on Windows 10.0.19044.1415:

* :ok: The Command Prompt session in the tab does not inherit the extra PATH.
* :ok: No `wsl.exe --list` process is started.
* :ok: The `wsl.exe -d Debian` process does not inherit the extra PATH.
* :warning: The Web browser process started by the Documentation link in the About dialog box inherits the extra path.
* :warning: The JSON file editor inherits the extra path.

**Comment by zadjii-msft** [comment]
Merging notes:

> I can reproduce this, but it is not specific to WSL.
> 
> 1. Pin Windows Terminal Preview 1.15.2002.0 to the task bar on Windows 10.0.19044.1826. (maintainer note: also repros on Win11)
> 2. Set Command Prompt as the default profile.
> 3. Close Windows Terminal Preview.
> 4. Start Windows Terminal Preview by left-clicking the task-bar button.
> 5. Execute `PATH` in the Command Prompt session. The output does not include the Windows Terminal Preview installation directory. ✅
> 6. Open another tab by right-clicking the task-bar button and left-clicking the "Command Prompt" task.
> 7. Execute `PATH` in the Command Prompt session in the second tab. The output does not include the Windows Terminal Preview installation directory. ✅
> 8. Close Windows Terminal Preview.
> 9. Start Windows Terminal Preview by right-clicking the task-bar button and left-clicking the "Command Prompt" task.
> 10. Execute `PATH` in the Command Prompt session. The output starts with Windows Terminal Preview installation directory. ❌
> 11. Open another tab by right-clicking the task-bar button and left-clicking the "Command Prompt" task.
> 12. Execute `PATH` in the Command Prompt session in the second tab. The output starts with Windows Terminal Preview installation directory. ❌
>

**Comment by zadjii-msft** [comment]
A thought: This might have been fixed by #14999.

**Comment by ianjoneill** [comment]
> A thought: This might have been fixed by #14999.

Yep - looks like it:

![image](https://user-images.githubusercontent.com/5821575/226851102-517daefe-65af-49e1-ac23-8b463bc33290.png)

And if I set `compatibility.reloadEnvironmentVariables` to `false`:

![image](https://user-images.githubusercontent.com/5821575/226851359-896a10fe-46ab-4460-9321-284d87279099.png)
